nav2_bringup/launch/nav2_bringup_launch.py

```diff
@@ -46,144 +61,87 @@ def generate_launch_description():
-        event_handler=launch.event_handlers.OnProcessExit(
-            target_action=start_rviz_cmd,
-            on_exit=launch.actions.EmitEvent(event=launch.events.Shutdown(reason='rviz exited'))))
-
-    start_map_server_cmd = launch.actions.ExecuteProcess(
-        cmd=[
-            os.path.join(
-                get_package_prefix('nav2_map_server'),
-                'lib/nav2_map_server/map_server'),
-            ['__params:=', params_file]],
-        cwd=[launch_dir], output='screen')
-
-    start_localizer_cmd = launch.actions.ExecuteProcess(
-        cmd=[
-            os.path.join(
-                get_package_prefix('nav2_amcl'),
-                'lib/nav2_amcl/amcl'),
-            ['__params:=', params_file]],
-        cwd=[launch_dir], output='screen')
-
-    start_world_model_cmd = launch.actions.ExecuteProcess(
-        cmd=[
-            os.path.join(
-                get_package_prefix('nav2_world_model'),
-                'lib/nav2_world_model/world_model'),
-            ['__params:=', params_file]],
-        cwd=[launch_dir], output='screen')
-
-    start_dwb_cmd = launch.actions.ExecuteProcess(
-        cmd=[
-            os.path.join(
-                get_package_prefix('dwb_controller'),
-                'lib/dwb_controller/dwb_controller'),
-            ['__params:=', params_file]],
-        cwd=[launch_dir], output='screen')
-
-    start_planner_cmd = launch.actions.ExecuteProcess(
-        cmd=[
-            os.path.join(
-                get_package_prefix('nav2_navfn_planner'),
-                'lib/nav2_navfn_planner/navfn_planner'),
-            ['__params:=', params_file]],
-        cwd=[launch_dir], output='screen')
-
-    start_navigator_cmd = launch.actions.ExecuteProcess(
-        cmd=[
-            os.path.join(
-                get_package_prefix('nav2_bt_navigator'),
-                'lib/nav2_bt_navigator/bt_navigator'),
-            ['__params:=', params_file]],
-        cwd=[launch_dir], output='screen')
-
-    start_lifecycle_manager_cmd = launch.actions.ExecuteProcess(
-        cmd=[
-            os.path.join(
-                get_package_prefix('nav2_lifecycle_manager'),
-                'lib/nav2_lifecycle_manager/lifecycle_manager'),
-            ['__params:=', params_file]],
-        cwd=[launch_dir], output='screen')
+    declare_autostart_cmd = launch.actions.DeclareLaunchArgument(
+        'autostart', default_value='true',
+        description='Automatically startup the nav2 stack')
+
+    declare_bt_xml_cmd = launch.actions.DeclareLaunchArgument(
+        'bt_xml_file',
+        default_value=os.path.join(get_package_prefix('nav2_bt_navigator'),
+            'behavior_trees', 'navigate_w_replanning_and_recovery.xml'),
+        description='Full path to the behavior tree xml file to use')
+
+    start_map_server_cmd = launch_ros.actions.Node(
+        package='nav2_map_server',
+        node_executable='map_server',
+        node_name='map_server',
```

Comment:
We shouldn't need the node_name option when we pass a params file as `parameters`. I'm not sure if it is still a problem, but it seems safer to leave it out if possible.

Reply:
I'm inclined to leave it in since I've validated it works. I'd want to re-test if I remove it.